mirror-service/src/main/java/org/acme/MirrorServiceTopology.java

```diff
@@ -0,0 +1,62 @@
+package org.acme;
+
+import io.quarkus.kafka.client.serialization.ObjectMapperSerde;
+import org.acme.common.KafkaTopic;
+import org.acme.model.OsvAdvisory;
+import org.acme.model.Vulnerability;
+import org.acme.osv.OsvAnalyzer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Named;
+import org.apache.kafka.streams.kstream.Produced;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import java.util.List;
+import java.util.UUID;
+
+import static org.acme.commonutil.KafkaStreamsUtil.processorNameConsume;
+import static org.acme.commonutil.KafkaStreamsUtil.processorNameProduce;
+
+@ApplicationScoped
+public class MirrorServiceTopology {
+
+    private final OsvAnalyzer osvAnalyzer;
+
+    @Inject
+    public MirrorServiceTopology(final OsvAnalyzer osvAnalyzer) {
+        this.osvAnalyzer = osvAnalyzer;
+    }
+
+    @Produces
+    public Topology topology() {
+
+        final var streamsBuilder = new StreamsBuilder();
+        final var osvAdvisorySerde = new ObjectMapperSerde<>(OsvAdvisory.class);
+
+        // OSV mirroring stream
+        // (K,V) to be consumed as (event uuid, list of ecosystems)
```

Comment:
As an alternative, you could use the ecosystem as message key.

Pros:

* Multiple ecosystems could be mirrored in parallel
* There would be no way the same ecosystem can be mirrored by multiple threads at the same time
* Less time is spent on processing a single input event

Cons:

* When N ecosystems are supposed to be mirrored, there would need to be N trigger events

Reply:
It's a good idea, to make it more grained. Pros > Cons 👍🏼 
So, API Server will trigger event for ecosystem enabled. 
Also, in that case, we don't have any value but just key (ecosystem string) in the event if I'm not wrong?